### DISCUSSION TITLE
Porting Broadcasters Functionality from Media Soup Demo V3 to Edumeet

### DISCUSSION THREAD

**Discussion by stemcc** [edumeet/edumeet#787] [Ideas]
Curious if there's been any discussion of possibly porting the broadcasters functionality from the latest mediasoup demo over to edumeet. 

The lionshare of the code is here: 

https://github.com/versatica/mediasoup-demo/blob/b5c668f8f474628fe7d7133b132fd265080a2d56/server/server.js

from roughly lines 130 - 420.

**Comment by stemcc** [comment]
Would a pull request be the best course here?  

If so, I'm happy to work on submitting one if the community feels like this is a helpful feature to port over from the mediasoup demo.

**Reply by misi** [reply]
Yes please make PR and we can try to integrate it. 
But please wait until the TS transition and RoomClient will be splitted in smaller more organized parts.

**Comment by so010** [comment]
Could you explain the use case? Which stream should be broadcasted to where for what purpose?

**Reply by stemcc** [reply]
Yes sure @so010! In this case, "broadcasting" means sending a media stream that originated outside of webrtc into the room. The use case is sending media from gstreamer/ffmpeg into a room that the peers can then consume like it was any normal connected peer. 

So for example, you could play out an MP4 file into the room (which is the example used in the shell scripts for ffmpeg/gstreamer in the mediasoup v3 demo). Other use cases would be sending rtp / rtmp / rtsp / hls / etc into the room so that room peers can watch the stream in sync together.

It basically is to allow using gstreamer/ffmpeg as a middleware that can "broadcast" into a room any type of media those tools can consume, just by leveraging mediasoup's plain RTP router.
